packages/sit-onyx/src/components/OnyxInput/OnyxInput.vue

```diff
@@ -92,7 +98,24 @@ const errorClass = useErrorClass(showError);
             :maxlength="props.maxlength"
             :aria-label="props.hideLabel ? props.label : undefined"
             :title="props.hideLabel ? props.label : undefined"
+            @focus="isFocused = true"
+            @blur="isFocused = false"
           />
+          <OnyxIcon
+            v-if="showClearIcon && isFocused && value !== ''"
+            class="onyx-clear-icon"
+            :icon="xSmall"
+            color="neutral"
+            @mousedown.prevent
+            @click="() => emit('update:modelValue', '')"
+          />
+
+          <OnyxIcon
+            v-if="!isFocused && showSuccessIcon && successMessages"
+            :icon="checkSmall"
+            color="success"
+          />
```

Comment:
Semantically, if the icons are clickable, they should be wrapped inside a `<button>` so they are also correctly usable via keyboard

Reply:
changed it